File: tests/src/testTransport/ActiveLink.cpp
#include "ActiveLink.h"

#include "Transport.h"

namespace testTransport {

ActiveLink::ActiveLink( Transport* creator, const std::string& address ) : _address( address ), 
    _creator( creator )
{
}

ActiveLink::ActiveLink()
{
    // empty constructor
}

ActiveLink::~ActiveLink()
{
    _creator->onLinkDestructor( _address );
}
    
// ------ reef.IActiveLink Methods ------ //

bool ActiveLink::receiveReply( std::string& msg )
{
    return _creator->checkReply( _address, msg );
}

void ActiveLink::send( const std::string& msg )
{
    _creator->post( _address, msg );
}


CORAL_EXPORT_COMPONENT( ActiveLink, ActiveLink );

} // namespace reef

File: modules/reef/rpc/ActiveLink.lua
local ActiveLink = co.Component( { name = "reef.rpc.ActiveLink", 
								provides = { active = "reef.rpc.IActiveLink" } } )

function ActiveLink:send( msg )
	self.net:send( self.address, msg )
	self.net:updateAll()
end

function ActiveLink:receiveReply()
	local msg = self.net:checkReply( self.address )
	
	if msg then
		return true, msg
	end
	
	self.net:updateAll()
	return false, ""
end

function ActiveLink:getAddress()
	return self.address
end

return ActiveLink